rllib/env/vector_env.py

```diff
@@ -321,14 +348,13 @@ def send_actions(self, action_dict: MultiEnvDict) -> None:
 
     @override(BaseEnv)
     def try_reset(self, env_id: Optional[EnvID] = None) -> MultiEnvDict:
-        from ray.rllib.env.base_env import _DUMMY_AGENT_ID
-
         assert env_id is None or isinstance(env_id, int)
-        return {
-            env_id
-            if env_id is not None
-            else 0: {_DUMMY_AGENT_ID: self.vector_env.reset_at(env_id)}
-        }
+        if not self.is_multiagent:
+            from ray.rllib.env.base_env import _DUMMY_AGENT_ID
+
+            return {env_id or 0: {_DUMMY_AGENT_ID: self.vector_env.reset_at(env_id)}}
```

Comment:
better to write with_dummy_agent_id({env_id or 0: self.vector_env.reset_at(env_id)})?